python/tvm/topi/tensor.py

```diff
@@ -73,3 +75,341 @@ def full_like(x, fill_value):
+        out = ib.buffer_ptr(out_buf)
+
+        temp_index = ib.allocate("int32", (data.shape[0]), name="temp_index", scope="local")
+        num = ib.allocate("int32", (1), name="num", scope="local")
+
+        shape = get_const_tuple(data.shape)
+        num_segment = get_const_tuple(out_buf.shape)[0]
+        inner_size = 1
+        for s in range(1, len(shape)):
+            inner_size = inner_size * shape[s]
+
+        with ib.for_range(0, num_segment) as n:
+            with ib.for_range(0, inner_size) as j:
+                out_index = n * inner_size + j
+                out[out_index] = 0.0
+
+            num[0] = 0
+            with ib.for_range(0, shape[0]) as k:
+                with ib.if_scope(seg_ids[k] == n):
+                    temp_index[num[0]] = k
+                    num[0] += 1
+
+            with ib.if_scope(num[0] > 0):
+                with ib.for_range(0, inner_size) as l:
+                    out_index = n * inner_size + l
+                    with ib.for_range(0, num[0]) as k:
+                        in_index = temp_index[k] * inner_size + l
+                        out[out_index] += input_data[in_index]
+                    out[out_index] = out[out_index] / num[0]
+
+        return ib.get()
+
+    assert len(segment_ids.shape) == 1
+
+    out_shape = list(get_const_tuple(data.shape))
+    out_shape[0] = num_out
+
+    out = te.extern(
+        out_shape,
+        [data, segment_ids],
+        lambda ins, outs: _segment_mean(ins[0], ins[1], outs[0]),
+        dtype=data.dtype,
+    )
+
+    return out
+
+
+def segment_sum(data, segment_ids, num_out):
+    """segment_sum operator.
+
+    Parameters
+    ----------
+    data : tvm.te.Tensor
+        input data
+
+    segment_ids : tvm.te.Tensor
+        input segment ids
+
+    num_out : int
+        number of output
+
+    Returns
+    -------
+    out : tvm.te.Tensor
+        Tensor with shape determined by the segment ids.
+    """
+
+    def _segment_sum(data, segment_ids, out_buf):
+
+        ib = tir.ir_builder.create()
+        input_data = ib.buffer_ptr(data)
+        seg_ids = ib.buffer_ptr(segment_ids)
+        out = ib.buffer_ptr(out_buf)
+
+        temp_index = ib.allocate("int32", (data.shape[0]), name="temp_index", scope="local")
+        num = ib.allocate("int32", (1), name="num", scope="local")
+
+        shape = get_const_tuple(data.shape)
+        num_segment = get_const_tuple(out_buf.shape)[0]
+        inner_size = 1
+        for s in range(1, len(shape)):
+            inner_size = inner_size * shape[s]
+
+        with ib.for_range(0, num_segment) as n:
+            with ib.for_range(0, inner_size) as j:
+                out_index = n * inner_size + j
+                out[out_index] = 0.0
+
+            num[0] = 0
+            with ib.for_range(0, shape[0]) as k:
+                with ib.if_scope(seg_ids[k] == n):
+                    temp_index[num[0]] = k
+                    num[0] += 1
+
+            with ib.if_scope(num[0] > 0):
```

Comment:
So, if the segment is not present we omit it? Why we don't do this for max and min? Some explanation of this could would be useful, I think

Reply:
At that time, it was to ensure that the division by 0 would not occur during the calculation process. Now modified to only be used in the `mean`